lib/superstore/types/float_type.rb

```diff
@@ -1,12 +1,8 @@
 module Superstore
   module Types
     class FloatType < BaseType
-      def decode(str)
-        str.to_f unless str.empty?
-      end
-
       def typecast(value)
-        value.to_f
+        Float(value) rescue nil
```

Comment:
Float coercion is used here rather than `to_f`, so that blank strings become nil rather than 0.0.